airflow/contrib/hooks/gcp_vision_hook.py

```diff
@@ -0,0 +1,254 @@
+    def create_product_set(self, location, product_set, project_id=None,
+                           product_set_id=None, retry=None, timeout=None, metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductSetCreateOperator`
+        """
+        client = self.get_client()
+        parent = PARENT_TEMPLATE.format(project_id, location)
+        self.log.info('Creating a new ProductSet under the parent: %s', parent)
+        response = self._handle_request(lambda **kwargs: client.create_product_set(**kwargs), parent=parent,
+                                        product_set=product_set, product_set_id=product_set_id, retry=retry,
+                                        timeout=timeout, metadata=metadata)
+        self.log.info('ProductSet created:\n%s', response)
+
+        if not product_set_id:
+            # Product set id was generated by the API
+            product_set_id = self._get_autogenerated_id(response)
+            self.log.info('Extracted autogenerated ProductSet ID from the response: %s', product_set_id)
+
+        return product_set_id
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def get_product_set(self, location, product_set_id, project_id=None,
+                        retry=None, timeout=None, metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductSetGetOperator`
+        """
+        client = self.get_client()
+        name = PRODUCT_SET_NAME_TEMPLATE.format(project_id, location, product_set_id)
+        self.log.info('Retrieving ProductSet: %s', name)
+        response = self._handle_request(lambda **kwargs: client.get_product_set(**kwargs), name=name,
+                                        retry=retry, timeout=timeout, metadata=metadata)
+        self.log.info('ProductSet retrieved:\n%s', response)
+        return MessageToJson(response)
+
+    @GoogleCloudBaseHook.fallback_to_default_project_id
+    def update_product_set(self, product_set, location=None, product_set_id=None, update_mask=None,
+                           project_id=None, retry=None, timeout=None, metadata=None):
+        """
+        For the documentation see:
+        :py:class:`~airflow.contrib.operators.gcp_vision_operator.CloudVisionProductSetUpdateOperator`
+        """
+        client = self.get_client()
+        product_set = self._determine_productset_name(product_set, product_set_id, location, project_id)
+        self.log.info('Updating ProductSet: %s', product_set.name)
+        response = self._handle_request(lambda **kwargs: client.update_product_set(**kwargs),
+                                        product_set=product_set, update_mask=update_mask, retry=retry,
+                                        timeout=timeout, metadata=metadata)
+        self.log.info('ProductSet updated:\n%s', response)
```

Comment:
The whole response should be at a lower logging level.

Reply:
Done